apps/ae_socket_connector/test/aesocketconnector_test.exs

```diff
@@ -1,8 +1,174 @@
 defmodule SocketConnectorTest do
   use ExUnit.Case
-  doctest SocketConnector
+  require ClientRunner
 
-  test "greets the world" do
-    assert SocketConnector.hello() == :world
+  @ae_url ClientRunner.ae_url()
+  @network_id ClientRunner.network_id()
+
+  def gen_names(id) do
+    clean_id = Atom.to_string(id)
+    {String.to_atom("alice " <> clean_id), String.to_atom("bob " <> clean_id)}
+  end
+
+  def custom_config(overide_basic_param, override_custom) do
+    fn initator_pub, responder_pub ->
+      %{basic_configuration: basic_configuration} =
+        Map.merge(ClientRunner.default_configuration(initator_pub, responder_pub), overide_basic_param)
+
+      %{
+        basic_configuration: basic_configuration,
+        custom_param_fun: fn role, host_url ->
+          Map.merge(ClientRunner.custom_connection_setting(role, host_url), override_custom)
+        end
+      }
+    end
+  end
+
+  def accounts_initiator() do
+    {TestAccounts.initiatorPubkeyEncoded(), TestAccounts.initiatorPrivkey()}
+  end
+
+  def accounts_responder() do
+    {TestAccounts.responderPubkeyEncoded(), TestAccounts.responderPrivkey()}
+  end
+
+  @tag :hello_world
+  test "hello fsm", context do
+    {alice, bob} = gen_names(context.test)
+
+    ClientRunner.start_peers(
+      @ae_url,
+      @network_id,
+      {alice, accounts_initiator()},
+      {bob, accounts_responder()},
+      &TestScenarios.hello_fsm_v2/3,
```

Comment:
Test scenarios should not be outside of this file, instead this file is meant to define those.